nemo/collections/asr/models/msdd_models.py

```diff
@@ -0,0 +1,1736 @@
+        """
+        Check whether the laoded clustering label file is including clustering results for all sessions.
+        This function is used for inference mode of MSDD.
+
+        Args:
+            out_dir (str):
+                Path to the directory where clustering result files are saved.
+
+        Returns:
+            file_exists (bool):
+                Boolean that indicates whether clustering result file exists.
+            clus_label_path (str):
+                Path to the clustering label output file.
+        """
+        clus_label_path = os.path.join(
+            out_dir, 'speaker_outputs', f'subsegments_scale{self.base_scale_index}_cluster.label'
+        )
+        file_exists = os.path.exists(clus_label_path)
+        if not file_exists:
+            logging.info(f"Clustering label file {clus_label_path} does not exist.")
+        return file_exists, clus_label_path
+
+    def load_clustering_labels(self, out_dir):
+        """
+        Load clustering labels generated by clustering diarizer. This function is used for inference mode of MSDD.
+
+        Args:
+            out_dir (str):
+                Path to the directory where clustering result files are saved.
+        Returns:
+            emb_scale_seq_dict (dict):
+                List containing clustering results in string format.
+        """
+        file_exists, clus_label_path = self.check_clustering_labels(out_dir)
+        logging.info(f"Loading cluster label file from {clus_label_path}")
+        with open(clus_label_path) as f:
+            clus_labels = f.readlines()
+        return clus_labels
+
+    def load_emb_scale_seq_dict(self, out_dir):
+        """
+        Load saved embeddings generated by clustering diarizer. This function is used for inference mode of MSDD.
+
+        Args:
+            out_dir (str):
+                Path to the directory where embedding pickle files are saved.
+        Returns:
+            emb_scale_seq_dict (dict):
+                Dictionary containing embedding tensors which are indexed by scale numbers.
+        """
+        window_len_list = list(self.cfg_base.diarizer.speaker_embeddings.parameters.window_length_in_sec)
+        emb_scale_seq_dict = {scale_index: None for scale_index in range(len(window_len_list))}
+        for scale_index in range(len(window_len_list)):
+            pickle_path = os.path.join(
+                out_dir, 'speaker_outputs', 'embeddings', f'subsegments_scale{scale_index}_embeddings.pkl'
+            )
+            logging.info(f"Loading embedding pickle file of scale:{scale_index} at {pickle_path}")
+            with open(pickle_path, "rb") as input_file:
+                emb_dict = pkl.load(input_file)
+            for key, val in emb_dict.items():
+                emb_dict[key] = val
+            emb_scale_seq_dict[scale_index] = emb_dict
+        return emb_scale_seq_dict
+
+
+class EncDecDiarLabelModel(ModelPT, ExportableEncDecModel, ClusterEmbedding):
```

Comment:
I don't see you are using any methods from ClusterEmbedding class, why are you inheriting it?

Reply:
Removed

Reply:
removed ClusterEmbedding